examples/extensions/authorizer.middleware.ts

```diff
@@ -0,0 +1,43 @@
+import { GraphQLResolveInfo, GraphQLObjectType } from "graphql";
+
+import { MiddlewareFn } from "../../src";
+import { Context } from "./context.interface";
+import { UnauthorizedError } from "../../src/errors";
+
+const extractAuthorizationExtensions = (info: GraphQLResolveInfo) => {
+  const parentAuthorizationExtensions =
+    (info.parentType.extensions && info.parentType.extensions.authorization) || {};
+  const returnType = info.returnType as GraphQLObjectType;
+  const returnTypeAuthorizationExtensions =
+    (returnType.extensions && returnType.extensions.authorization) || {};
+  const field = info.parentType.getFields()[info.fieldName];
+  const fieldAuthorizationExtensions = (field.extensions && field.extensions.authorization) || {};
+
+  return {
+    ...parentAuthorizationExtensions,
+    ...returnTypeAuthorizationExtensions,
+    ...fieldAuthorizationExtensions,
+  };
+};
+
+export const AuthorizerMiddleware: MiddlewareFn = async (
```

Comment:
I have chosen to base my examples off of the "authorization" examples, and showcase how we could implement it in a custom way with extensions, I hope it makes sense?

Reply:
I had already added the logger middleware in the examples as well 🙂 `examples/extensions/logger.middleware.ts`

My idea was to show we can define custom decorators on top of @Extensions with one middleware, but also that we can use @Extensions directly with another middleware. But I can also combine both approaches with one middleware 🙂I attempted to do it in this commit: 6617818ad896965a51fa55500d5a0ca2147112f8, I removed the whole authorization thing and only kept and extended the logger example. 

What do you think? I can also revert it if the previous iteration looked better

Reply:
Looks ok I think 😉